File: CESMAC 4° periodo/POO/AV1/ESTUDOS/CESMAC/aula06-packed/atividade/modulo/Produto.py
class Produto:
    def __init__(self,nome,preco,estoque):
        self._nome = nome
        self._preco = preco
        self._estoque = estoque


    def get_nome(self):
        return self._nome

    def get_preco(self):
        return self._preco

    def get_estoque(self):
        return self._estoque

    def set_nome(self,nome):
        self._nome = nome

    def set_preco(self,preco):
        self._preco = preco

    def set_estoque(self,estoque):
        self._estoque = estoque

File: CESMAC 4° periodo/POO/AV1/ESTUDOS/CESMAC/aula06-packed/atividade/modulo/Produto.java
package modulo;

public class Produto {
  
    private String nome;
    private double preco;
    private int estoque;
    
    public Produto(String nome, double preco, int estoque) {
        this.nome = nome; 
        this.preco = preco; 
        this.estoque = estoque; 
    }

    // Encapsulamento - Getters e Setters
    
    // Nome
    public String getNome(){
        return nome;
    }
    
    public void setNome(String nome){
        this.nome = nome;
    }

    // Preço
    public double getPreco(){
        return preco;
    }

    public void setPreco(double preco){
        this.preco = preco;
    }
    
  
    
    // Estoque
    public int getEstoque(){
        return estoque;
    }
    
    public void setEstoque(int estoque){
        this.estoque = estoque;
    }
   
    
    public void aplicarDesconto() {
        this.preco -= 10; 
    }

    public void reduzirEstoque() {
        this.estoque -= 1; 
    }
    
    public void exibirInformacoes() {
        System.out.println("____INFORMAÇÕES____");
        System.out.println("Produto: " + this.nome);  
        System.out.println("Preço: " + this.preco); 
        System.out.println("Estoque: " + this.estoque); 
    }
}





